### PR TITLE
Finish StorageMarketActor open items

### PR DESCRIPTION
**Pull request by jzimmerman**
- Generate storage deal IDs
- Placeholders for partial slash and other parameters
- Handle non-miner (client) callers
- Finish caller validation
- Minor edits and cleanup

### CODE REVIEW

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go [hunk 1]
```diff
@@ -252,6 +245,9 @@ func (st *StorageMarketActorState_I) _rtUpdatePendingDealStatesForParty(rt Runti
 	var relevantDealIDs []deal.DealID
 	// Populate with the set of all elements in st.Deals() in which addr is one of the parties
 	// (either client or provider).
+	//
+	// Note: as an optimization, implementations may cache efficient data structures to maintain
+	// this index.
```
Comment (anorth) [inline src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: No! In a real VM, that would change the operations and hence the gas cost. Define the operations here.

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go [hunk 2]
```diff
@@ -299,6 +301,28 @@ func _rtAbortIfDealInvalidForNewSectorSeal(
...
+	}
+}
+
+func _rtIsStorageMiner(rt Runtime, minerAddr addr.Address) bool {
+	codeID, ok := rt.GetActorCodeID(minerAddr)
+	Assert(ok)
+	if !codeID.IsBuiltin() {
+		return false
+	}
+	return (codeID.As_Builtin() == actor.BuiltinActorID_StorageMiner)
```
Comment (anorth) [inline src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: No, this is so brittle! I object in the strongest way. If the storage market needs different behaviour for different kinds of accounts, it should maintain state differentiating those accounts.

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go [hunk 3]
```diff
@@ -7,82 +7,60 @@ import (
...
-func (a *StorageMarketActorCode_I) AddBalance(rt Runtime, minerAddr addr.Address) {
-	TODO() // TODO: must also handle case of non-miner (client) sender
-
-	ownerAddr, workerAddr := _rtGetMinerAccountsAssert(rt, minerAddr)
-	rt.ValidateImmediateCallerInSet([]addr.Address{ownerAddr, workerAddr})
-	TODO() // TODO: should workerAddr be permitted here? Depends on actor principal security assumptions.
-
-	msgValue := rt.ValueReceived()
+func (a *StorageMarketActorCode_I) AddBalance(rt Runtime, entryAddr addr.Address) {
+	_rtValidateImmediateCallerDetermineRecipient(rt, entryAddr)
```
Comment (anorth) [inline src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: In conversation we clearly decided that anyone could add balance to an account. This should not do any caller verification at all.

File: src/systems/filecoin_vm/indices/indices.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+func (inds *Indices_I) StorageDeal_ClientCollateralBounds(
+	pieceSize piece.PieceSize,
+	startEpoch block.ChainEpoch,
+	endEpoch block.ChainEpoch,
+) (minClientCollateral actor.TokenAmount, maxClientCollateral actor.TokenAmount) {
+
+	// placeholder
+	PARAM_FINISH()
+	panic("")
+}
```
Comment (anorth) [inline src/systems/filecoin_vm/indices/indices.go]: Objection: this is all simple business logic. It does not belong in the VM/runtime at all.

File: src/systems/filecoin_vm/runtime/impl/runtime.go [hunk 5]
```diff
@@ -363,7 +363,7 @@ func (rt *VMContext) _transferFunds(from addr.Address, to addr.Address, amount a
 	return nil
 }
 
-func (rt *VMContext) _getActorCodeID(actorAddr addr.Address) (ret actor.CodeID, ok bool) {
+func (rt *VMContext) GetActorCodeID(actorAddr addr.Address) (ret actor.CodeID, ok bool) {
```
Comment (anorth) [inline src/systems/filecoin_vm/runtime/impl/runtime.go]: Objection! Actor have no good reason to inspect the code type of another actor.

**Comment by jzimmerman** [comment]
(Noting for posterity: comments above have since been addressed via out-of-band discussions.)

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go (hunk 1 above)
Comment (jzimmerman) [reply src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: (Addressed via out-of-band discussions.)

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go (hunk 3 above)
Comment (jzimmerman) [reply src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: (Addressed via out-of-band discussions.)

File: src/systems/filecoin_markets/storage_market/storage_market_actor_code.go (hunk 2 above)
Comment (jzimmerman) [reply src/systems/filecoin_markets/storage_market/storage_market_actor_code.go]: (Addressed via out-of-band discussions.)

File: src/systems/filecoin_vm/indices/indices.go (hunk 4 above)
Comment (jzimmerman) [reply src/systems/filecoin_vm/indices/indices.go]: (Addressed via out-of-band discussions.)

File: src/systems/filecoin_vm/runtime/impl/runtime.go (hunk 5 above)
Comment (jzimmerman) [reply src/systems/filecoin_vm/runtime/impl/runtime.go]: (Addressed via out-of-band discussions.)

**Pull request merged**